### ISSUE TITLE
[🐛] [Android] Support 16kb page size for Android 15+ devices

### ISSUE THREAD

**Issue by k-saparia** [invertase/react-native-firebase#7954]
### Description
I am encountering issues when running my app on Android 15 devices with a 16 KB page size, particularly with some native components. Are there any plans to recompile the library to address these issues and ensure compatibility with phones that have a 16 KB page size?

As stated on Android [doc](https://developer.android.com/guide/practices/page-sizes#compile-16-kb-alignment),

> Beginning with Android 15, Android supports devices that are configured to use a page size of 16 KB (16 KB devices). If your app uses any [NDK](https://developer.android.com/ndk) libraries, either directly or indirectly through an SDK, then you will need to rebuild your app for it to work on these 16 KB devices.

An analysis of my app using the APK Analyzer [as described by Android](https://developer.android.com/guide/practices/page-sizes#16-kb-impact) shows that it does use **libcrashlytics-common.so** as native code.
<img width="1044" alt="image" src="https://github.com/user-attachments/assets/9917fbf1-b50c-4366-bdb1-a49234a5a2a2">
This is the native binary reported by the APK analysis:

libcrashlytics-common.so

### Describe the solution you'd like
I would like for **libcrashlytics-common** to support 16kb page size phones so that apps supporting sdk 35 do not crash
### Additional context
Android doc: https://developer.android.com/guide/practices/page-sizes#compile-16-kb-alignment
react-native thread: https://github.com/facebook/react-native/issues/45054
### What Platforms whould this feature/enhancement affect?
Android

### Alternatives/Workarounds
No workaround I have investigated so far.

**Comment by russellwheatley** [comment]
Based on the issue you created here: https://github.com/firebase/firebase-android-sdk/issues/6142

android sdk support is in firebase-android-sdk => 33.1.1.

which has already been released in the latest version: https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#bug-fixes-2

**Comment by chrispader** [comment]
Hey 👋🏼 Chiming back in on this one, React Native Version 0.77 with 16KB memory page size support has been released a while ago.

Referring to [these instructions](https://developer.android.com/guide/practices/page-sizes#build), I've just checked and i don't see any usages of  `PAGE_SIZE`, `getpagesize()` or `sysconf()` on the first look. So therefore, i think the only needed change should be adding the following to the `android/build.gradle` file:

```groovy
android {
  defaultConfig {
    externalNativeBuild {
      cmake {
        arguments "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
      }
    }
  }
}
```

@mikehardy let me know if you think there are further changes to be made. I'd be glad to help adding support for this one! 🙌🏼
